src/new_hope/week3.clj

```diff
@@ -0,0 +1,148 @@
+(ns new-hope.week3
+  (:require [midje.sweet :refer :all])
+  (:require [clojure.test :refer :all])
+  (:require [clojure.set :refer :all])
+  )
+
+;Day 1
+(defn is-binary-tree
+  "determines whether the argument sequence is a binary tree or not"
+  [x]
+  (if (not (coll? x))
+    (if (= x false) false true)
+    (if (= (count x) 3)
+      (and (is-binary-tree (first (rest x))) (is-binary-tree (last x)))
+      false))
+  )
+
+(deftest test-is-binary-tree
+  (is (= (is-binary-tree '(:a (:b nil nil) nil)) true))
+  (is (= (is-binary-tree '(:a (:b nil nil))) false))
+  (is (= (is-binary-tree [1 nil [2 [3 nil nil] [4 nil nil]]]) true))
+  (is (= (is-binary-tree [1 [2 nil nil] [3 nil nil] [4 nil nil]]) false))
+  (is (= (is-binary-tree [1 [2 [3 [4 nil nil] nil] nil] nil]) true))
+  (is (= (is-binary-tree [1 [2 [3 [4 false nil] nil] nil] nil]) false))
+  (is (= (is-binary-tree '(:a nil ())) false))
+  )
+
+(defn is-symmetric-tree
+  "determines whether the argument sequence is a binary tree or not"
+  [x y]
+  (if (and (coll? x) (coll? y))
+    (and (= (first x) (first y))
+         (and
+           (is-symmetric-tree (first (rest x)) (last y))
+           (is-symmetric-tree (last x) (first (rest y)))
+           ))
+    (= x y))
+  )
+
+(defn is-symetric-tree-self
+  [x]
+  (if (coll? x)
+    (is-symmetric-tree (first (rest x)) (last x))
+    true
+    )
+  )
+
+(deftest test-is-symmetric-tree
+  (is (= (is-symetric-tree-self '(:a (:b nil nil) (:b nil nil))) true))
+  (is (= (is-symetric-tree-self '(:a (:b nil nil) nil)) false))
+  (is (= (is-symetric-tree-self '(:a (:b nil nil) (:c nil nil))) false))
+  (is (= (is-symetric-tree-self [1 [2 nil [3 [4 [5 nil nil] [6 nil nil]] nil]]
+                                 [2 [3 nil [4 [6 nil nil] [5 nil nil]]] nil]]) true))
+  (is (= (is-symetric-tree-self [1 [2 nil [3 [4 [5 nil nil] [6 nil nil]] nil]]
+                                 [2 [3 nil [4 [5 nil nil] [6 nil nil]]] nil]]) false))
+  (is (= (is-symetric-tree-self [1 [2 nil [3 [4 [5 nil nil] [6 nil nil]] nil]]
+                                 [2 [3 nil [4 [6 nil nil] nil]] nil]]) false))
+  )
+
+;Day	2
+
+(defn flip-order
+  [x]
+  (fn [a b] (x b a))
+  )
+
+(deftest test-flip-order
+  (is (= 3 ((flip-order nth) 2 [1 2 3 4 5])))
+  (is (= true ((flip-order >) 7 8)))
+  (is (= 4 ((flip-order quot) 2 8)))
+  (is (= [1 2 3] ((flip-order take) [1 2 3 4 5] 3)))
+  )
+
+(defn rotate-a-sequence
+  [rotation collection]
+  (let [size (count collection) absrot (Math/abs rotation)]
+    (take-last size (take (if (pos? rotation) (+ size absrot) (+ size absrot 1)) (cycle collection))))
```

Comment:
Too complicated, hard to read.
Hint: use `mod`, `drop`, `take`. Don't use `Math/abs`.